tests/test_failsafe.py

```diff
@@ -118,6 +121,23 @@ def test_retry_on_custom_exception(self):
 
         assert failing_operation.called == retries + 1
 
+    def test_delay(self):
+        failing_operation = create_failing_operation()
+        retries = 1
+        delay = Delay(timedelta(seconds=0.2))
+        policy = RetryPolicy(retries, [SomeRetriableException], backoff=delay)
+        with pytest.raises(RetriesExhausted):
+            loop.run_until_complete(
+                Failsafe(retry_policy=policy)
+                .run(failing_operation)
+            )
+
+    def test_backoff(self):
+        retries = 3
+        backoff = Backoff(timedelta(seconds=0.2), timedelta(seconds=1))
+        policy = RetryPolicy(retries, [SomeRetriableException], backoff=backoff)
+        Failsafe(retry_policy=policy)
+
```

Comment:
I might be missing something, but I don't see any of these two tests actually verify anything. One suggestion: patch `asyncio.sleep`, and check that it is invoked with the expected value, the amount of expected times. Also, this way the test won't incurr in the unnecessary delay of actually calling `sleep()`.

Reply:
Almost there! Although assigning a mock directly instead of patching is bad practice, as it breaks test isolation (imagine, another test executes afterwards that relies on the actual`asyncio.sleep` and gets a mock instead). You coud just decorate the test with `@unittest.mock.patch('asyncio.sleep', get_mock_coro(None))`, this way you would preserve test isolation. More info about patching here: https://docs.python.org/3/library/unittest.mock.html#the-patchers

Reply:
Also, it would be nice to have the same kind of assertions in `test_delay`, wouldn't it?